lib/api/src/sys/instance.rs

```diff
@@ -121,18 +121,18 @@ impl Instance {
     /// Those are, as defined by the spec:
     ///  * Link errors that happen when plugging the imports into the instance
     ///  * Runtime errors that happen when running the module `start` function.
-    pub fn new(
+    pub fn new<'a>(
         module: &Module,
-        resolver: &(dyn Resolver + Send + Sync),
+        resolver: &'a (dyn Resolver + 'a),
```

Comment:
can be `&'a dyn Resolver` I think (due to lifetime elision rules for trait objects)?